### PR TITLE
Add endpoint to get Category details

### PR DESCRIPTION
**Pull request by SunandanBose**

### CODE REVIEW

File: scraper/aws.js [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
-    console.log("Inside saveToS3() ");
+async function save(data, type) {
+    console.log("Inside save() ");
+
+
     const s3Bucket = "stock-ui-bucket";
-    const objectName = "data/" + data.stockId;
     const objectType = "application/json";
+    var objectName = "";
+
```
Comment (kumarRaj) [inline scraper/aws.js:11]: Try to not have logic in this file

File: backend/src/main/java/com/project/stocks/repository/CategoryRepository.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Scanner;
+
+@Component
+public class CategoryRepository {
+
+    public Category getCategoryDetails(String categoryName) {
+        final AmazonS3 s3 = AmazonS3ClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
+        Category category = new Category();
+        try {
+            String key_name = "category/" + categoryName;
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/repository/CategoryRepository.java:27]: S3 logic should not be here

File: backend/src/main/java/com/project/stocks/repository/CategoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            e.printStackTrace();
+        }
+        return category;
+    }
+
+    public CategoryList getCategoryNames() {
+        final AmazonS3 s3 = AmazonS3ClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
+        CategoryList categoryList = new CategoryList();
+        try {
+            String key_name = "categoryList";
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/repository/CategoryRepository.java:52]: same as above

File: backend/src/main/java/com/project/stocks/dto/CategoryList.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.List;
+
+public class CategoryList {
+
+    public CategoryList() {
+    }
+
+    @JsonProperty("category")
+    private List<String> category;
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/dto/CategoryList.java:13]: This class is a list of strings. Does not need to exist

**Review by kumarRaj — CHANGES_REQUESTED** [review]
Have given comments

File: scraper/aws.js (hunk 1 above)
Comment (SunandanBose) [reply scraper/aws.js]: Done

File: backend/src/test/java/com/project/stocks/service/DataMapperTest.java [hunk 5]
```diff
@@ -41,12 +42,37 @@ void testMapMethodInCaseValidDataShouldReturnStockObjectWithValidValues() throws
...
+    void testMapCategoryMethodInCaseOfEmptyInputShouldReturnNull() throws JsonProcessingException {
+        assertNull(DataMapper.mapCategory(""));
+    }
+
+    @Test
+    void testMapCategoryMethodInCaseValidDataShouldReturnStockObjectWithValidValues() throws JsonProcessingException {
+        String input = "{\"name\":\"NIFTY50\",\"company\":[\"ASIANPAINT\",\"GRASIM\",\"SBILIFE\",\"SBIN\",\"BPCL\",\"TCS\"," +
+                "\"JSWSTEEL\",\"M&M\",\"BAJAJFINSV\",\"HDFCLIFE\",\"ICICIBANK\",\"IOC\",\"ONGC\",\"ITC\",\"HEROMOTOCO\",\"NTPC\"," +
+                "\"BAJAJ-AUTO\",\"TATASTEEL\",\"SUNPHARMA\",\"RELIANCE\",\"BRITANNIA\",\"ULTRACEMCO\",\"BAJFINANCE\",\"SHREECEM\",\"MARUTI\"," +
+                "\"TATAMOTORS\",\"CIPLA\",\"HDFCBANK\",\"DRREDDY\",\"INDUSINDBK\",\"ADANIPORTS\",\"HINDUNILVR\",\"HINDALCO\",\"NESTLEIND\"" +
```
[the remark this answers is not in the document]
Comment (SunandanBose) [reply backend/src/test/java/com/project/stocks/service/DataMapperTest.java]: Done

File: backend/src/main/java/com/project/stocks/repository/CategoryRepository.java (hunk 2 above)
Comment (SunandanBose) [reply backend/src/main/java/com/project/stocks/repository/CategoryRepository.java:27]: Done

File: backend/src/main/java/com/project/stocks/repository/CategoryRepository.java (hunk 3 above)
Comment (SunandanBose) [reply backend/src/main/java/com/project/stocks/repository/CategoryRepository.java:39]: Done

File: backend/src/main/java/com/project/stocks/dto/CategoryList.java (hunk 4 above)
Comment (SunandanBose) [reply backend/src/main/java/com/project/stocks/dto/CategoryList.java]: Done

File: backend/src/main/java/com/project/stocks/controller/CategoryController.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/categories")
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/controller/CategoryController.java:14]: can be changed to /stock/categories

File: backend/src/main/java/com/project/stocks/dto/Category.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.List;
+
+public class Category {
+
+    @JsonProperty("name")
+    private String name;
+
+    @JsonProperty("company")
+    private List<String> companyList;
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/dto/Category.java:14]: name should be companies

File: backend/src/main/java/com/project/stocks/model/Score.java [hunk 8]
```diff
@@ -3,6 +3,8 @@
 public class Score {
 
     private int score = 0;
+    public final int total = 30;
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/model/Score.java:6]: Add comments around why this is 30

File: backend/src/main/java/com/project/stocks/repository/CategoryRepository.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+@Component
+public class CategoryRepository {
+
+    private final String bucketName = "stock-ui-bucket";
+
+    private S3AWSRepository s3AWSRepository;
+
+    @Autowired
+    public CategoryRepository(S3AWSRepository s3AWSRepository) {
+        this.s3AWSRepository = s3AWSRepository;
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/repository/CategoryRepository.java:21]: Could be called S3Repository

File: backend/src/main/java/com/project/stocks/repository/S3AWSRepository.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component
+public class S3AWSRepository {
+
+    private final String bucketName = "stock-ui-bucket";
+
+    public String getData(String bucketName, String key) {
+        String result = "";
+        try {
+            final AmazonS3 s3 = AmazonS3ClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/repository/S3AWSRepository.java:21]: Let's move region to application.properties

File: backend/src/main/java/com/project/stocks/service/DataMapper.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.project.stocks.dto.Category;
+import com.project.stocks.dto.Stock;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataMapper {
+
+    public static Stock mapStock(String result) throws JsonProcessingException {
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/service/DataMapper.java:14]: Combine all 3 functions

File: scraper/index.js [hunk 12]
```diff
@@ -101,4 +102,4 @@ function getOtherLiabilities(html) {
   return otherLiability
 }
 
-// testHandler("TCS");
+testHandler("TCS");
```
Comment (kumarRaj) [inline scraper/index.js:105]: is this intentinal?

**Review by kumarRaj — APPROVED** [review]

**Pull request merged**